omp/components/device_matrix_data_kernels.cpp

```diff
@@ -50,43 +50,51 @@ namespace components {
 
 template <typename ValueType, typename IndexType>
 void remove_zeros(std::shared_ptr<const DefaultExecutor> exec,
-                  Array<matrix_data_entry<ValueType, IndexType>>& data)
+                  Array<ValueType>& values, Array<IndexType>& row_idxs,
+                  Array<IndexType>& col_idxs)
```

Comment:
I think this might be a candidate for using plain pointers + size instead of wrapping in arrays ?

Reply:
Here I need to resize the arrays, so that's not possible unfortunately.

Reply:
Ah, I overlooked the resize.